src/aind_codeocean_utils/models/config.py

```diff
@@ -0,0 +1,161 @@
+"""Models for config files."""
+from typing import Dict, List, Optional
+
+from pydantic import BaseModel, Field, model_validator
+
+
+class RegisterDataConfig(BaseModel):
+    """
+    Settings for registering data
+    """
+
+    asset_name: str = Field(
+        ..., description="The name to give the data asset."
+    )
+    mount: str = Field(..., description="The mount folder name.")
+    bucket: str = Field(
+        ..., description="The s3 bucket the data asset is located."
+    )
+    prefix: str = Field(
+        ..., description="The s3 prefix where the data asset is located."
+    )
+    public: bool = Field(
+        default=False, description="Whether the data asset is public or not."
+    )
+    keep_on_external_storage: bool = Field(
+        default=True,
+        description="Whether to keep the data asset on external storage.",
+    )
+    tags: Optional[List[str]] = Field(
+        default=None, description="The tags to use to describe the data asset."
+    )
+    custom_metadata: Optional[Dict] = Field(
+        default=None,
+        description="What key:value metadata tags to apply to the asset.",
+    )
+    viewable_to_everyone: bool = Field(
+        default=False,
+        description="Whether to share the captured results with everyone.",
+    )
+
+    @model_validator(mode="after")
+    def check_tags_and_custom_metadata(self) -> "RegisterDataConfig":
+        """Check that tags and custom_metadata are lists and dicts"""
+        if self.tags is None:
+            self.tags = []
+        if self.custom_metadata is None:
+            self.custom_metadata = {}
+        return self
+
+
+class RunCapsuleConfig(BaseModel):
+    """
+    Settings for running a capsule
+    """
+
+    capsule_id: Optional[str] = Field(
+        default=None, description="ID of the capsule to run."
+    )
+    pipeline_id: Optional[str] = Field(
+        default=None, description="ID of the pipeline to run."
+    )
+    data_assets: Optional[List[Dict]] = Field(
```

Comment:
There is a `ComputationDataAsset` data class which has id and mount fields. It might be better to have
`data_assets: List[ComputationDataAsset]`
and have the validator coerce dictionaries into that data class.